File: 2358.cs
using System.Linq;
public class Solution {
    public int MaximumGroups(int[] grades) {
        Array.Sort(grades);

        int totalStudents = grades.Length;
        int groupCount = 0;
        int currentSize = 1;

        while (totalStudents >= currentSize)
        {
            totalStudents -= currentSize;
            groupCount++;
            currentSize++;
        }

        return groupCount;
    }
}

File: 2358.js
/**
 * @param {number[]} grades
 * @return {number}
 */
var maximumGroups = function(grades) {

    grades.sort((a, b) => a - b);
    let totalStudents = grades.length;
    let groupCount = 0;
    let currentSize = 1;

    while (totalStudents >= currentSize) {
        totalStudents -= currentSize;
        groupCount++;
        currentSize++;
    }

    return groupCount;
};
